src/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -212,19 +210,16 @@ app.directive('dashboardApp', function ($injector) {
       $scope.$watch('model.timeRestore', () => dashboardStateManager.setTimeRestore($scope.model.timeRestore));
       $scope.indexPatterns = [];
 
-      $scope.registerPanelIndexPattern = (panelIndex, pattern) => {
-        dashboardStateManager.registerPanelIndexPatternMap(panelIndex, pattern);
-        $scope.indexPatterns = dashboardStateManager.getPanelIndexPatterns();
-      };
-
       $scope.onPanelRemoved = (panelIndex) => {
         dashboardStateManager.removePanel(panelIndex);
         $scope.indexPatterns = dashboardStateManager.getPanelIndexPatterns();
       };
 
       $scope.$watch('model.query', $scope.updateQueryAndFetch);
 
-      $scope.$listen(timefilter, 'fetch', $scope.refresh);
+      $scope.$listen(timefilter, 'fetch', () => {
+        dashboardStateManager.handleTimeChange($scope.timefilter);
```

Comment:
We can't yet get rid of scope.refresh and the `$rootScope.$broadcast('fetch');` call because changing filters/query still goes through there.  Once we also shuffle that logic through to the embeddables directly (as desired for https://github.com/elastic/kibana/issues/16641) we should be able to get rid of this crazy, fragile async logic.

Reply:
Ah I was wrong about this, discover/embedded search needs the broadcast fetch call because it's not listening directly to the time filter passed down by the dashboard - but instead to the global one.  I added a test to catch the bug caused by removing this.